Content.Server/GameObjects/Components/Items/RCDComponent.cs

```diff
@@ -0,0 +1,169 @@
+                    break;
+                case RCDmode.Walls:
+                    _outputTile = "base_wall";
+                    break;
+                case RCDmode.Deconstruct:
+                    _outputTile = "space";
+                    break;
+            }
+            _serverNotifyManager.PopupMessage(Owner, eventArgs.User, "The RCD is now set to "+this.mode+" mode."); //Prints an overhead message above the RCD
+        }
+
+        /**
+         *
+         * Method called when the user examines this object, it'll simply add the mode that it's in to the object's description
+         * @params message = The original message from examining, like ..() in BYOND's examine
+         *
+         */
+
+        void IExamine.Examine(FormattedMessage message)
+        {
+            message.AddMarkup(Loc.GetString("It's currently in "+mode+" mode, and holds "+this.ammo+" charges."));
+        }
+
+        /**
+         *
+         * Method to handle clicking on a tile to then appropriately RCD it. This can have several behaviours depending on mode.
+         * @param eventAargs = An action event telling us what tile was clicked on. We use this to exrapolate where to place the new tile / remove the old one etc.
+         *
+         */
+
+        public void AfterAttack(AfterAttackEventArgs eventArgs)
+        {
+            var attacked = eventArgs.Attacked;
+            var mapGrid = _mapManager.GetGrid(eventArgs.ClickLocation.GridID);
+            var tile = mapGrid.GetTileRef(eventArgs.ClickLocation);
+
+            var coordinates = mapGrid.GridTileToLocal(tile.GridIndices);
+            float distance = coordinates.Distance(_mapManager, Owner.Transform.GridPosition);
+
+            if (distance > InteractionSystem.InteractionRange || ammo <= 0)
+            {
+                return;
+            }
+
+            var targetTile = (ContentTileDefinition) _tileDefinitionManager[tile.Tile.TypeId];
+            ITileDefinition desiredTile = null;
+            //Try catch to assert that _outputTile is a valid key in the tiledefinition manager, failing this check, we'll abort any tile placement.
+            try
+            {
+                desiredTile = _tileDefinitionManager[_outputTile];
+            }
+            catch (Exception e)
+            {
+                desiredTile = null;
+            }
+
+            var canPlaceTile = targetTile.IsSubFloor && desiredTile != null; //Boolean to check if we're able to build the desired tile. This defaults to checking for subfloors, but is overridden by "deconstruct" which sets it to the inverse.
+
+
+            switch (this.mode)
+            {
+                //Floor mode just needs the tile to be a space tile (subFloor)
+                case RCDmode.Floors:
+                    break;
+                //We don't want to place a space tile on something that's already a space tile. Let's do the inverse of the last check.
+                case RCDmode.Deconstruct:
```

Comment:
Can't this not deconstruct walls like this?

Reply:
Nope, haven't dug around for the correct method for removing an object yet

Reply:
`entity.Delete()` is what you want.